tests/test_parser.py

```diff
@@ -438,16 +438,16 @@ def test_setter(self, dfparser, instruction, existing, new, expected):
     @pytest.mark.parametrize(('old_instructions', 'key', 'new_value', 'expected'), [
         # Simple case, no '=' or quotes
         ('Release 1', 'Release', '2', 'Release 2'),
-        # No '=' but quotes
-        ('"Release" "2"', 'Release', '3', 'Release 3'),
+        # No '=' but quotes (which are kept)
+        ('"Release" "2"', 'Release', '3', '"Release" 3'),
         # Simple case, '=' but no quotes
         ('Release=1', 'Release', '6', 'Release=6'),
         # '=' and quotes
-        ('"Name"=\'alpha alpha\' Version=1', 'Name', 'beta delta', 'Name=\'beta delta\' Version=1'),
+        ('"Name"=\'alpha alpha\' Version=1', 'Name', 'beta delta', '"Name"=\'beta delta\' Version=1'),
         # '=', multiple labels, no quotes
         ('Name=foo Release=3', 'Release', '4', 'Name=foo Release=4'),
         # '=', multiple labels and quotes
-        ('Name=\'foo bar\' "Release"="4"', 'Release', '5', 'Name=\'foo bar\' Release=5'),
+        ('Name=\'foo bar\' "Release"="4"', 'Release', '5', 'Name=\'foo bar\' "Release"=5'),
```

Comment:
maybe add as well test:

('Release=1', 'Release', 'new value', "Release='new value'"),

Reply:
and 

('Release 1', 'Release', 'new value', "Release 'new value'")

Reply:
Thanks, this found a bug. :-)

Reply:
Just pushed another commit to fix this. Put it in its own commit for now for ease of review; will squash before merge.